core/static/core/js/search-results.js

```diff
@@ -132,20 +129,44 @@ function setup_merge_btn() {
     }
 });
 
-search_controls.forEach((target_id) => {
-    if(document.getElementById(target_id)) {
-            document.getElementById(target_id).addEventListener('change', () => {
-            submit_page();
+$('.searchcontrol').each((i, searchcontrol) => {
+    $(searchcontrol).on('change', (e) => {
+        // Every time a search control is changed, that change is persisted
+        let element_name = $(e.target).attr('name');
+        localStorage.setItem(`${entity}_${element_name}`, $(e.target).val());
+        //submit_page();
 
-        });
+    });
+
+    let searchParams = new URLSearchParams(window.location.search);
+
+    // Do not use persisted search control settings if they are explicitly set as
+    // get parameters
+    if((!searchParams.has(searchcontrol.name) || searchParams.get(searchcontrol.name) === '')
+     && localStorage.getItem(`${entity}_${searchcontrol.name}`) != null) {
+        // Set form values to persisted value
+        let val = localStorage.getItem(`${entity}_${searchcontrol.name}`);
+
+        if(searchcontrol.type != 'radio')   {
+            $(searchcontrol).val(val);
+        }
+        else    {
+            if(searchcontrol.value === val)   {
+                searchcontrol.checked = true;
+            }
+            else {
+                searchcontrol.checked = false;
+            }
+        }
     }
+
 });
 
 
 function setup_discard_page_on_new_search() {
     document.getElementById('search_form').addEventListener('change', function(e) {
         // No need to mark if search controls change
-        if(search_controls.indexOf(e.srcElement.id) == -1)   {
+        if($(e.srcElement).attr('class') != 'searchcontrol')   {
             e.srcElement.dataset['changed'] =  true;
```

Comment:
is it possible to use `target` instead of `srcElement`

![image](https://user-images.githubusercontent.com/97410502/230036628-071dc3cf-8f78-41f9-94f1-c78bbe4dedd7.png)